sdk/core/azure-core/src/main/java/com/azure/core/models/spatial/Point.java

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.models.spatial;
+
+import com.azure.core.annotation.Immutable;
+
+/**
+ * Represents a location in (x, y) coordinate space.
+ */
+@Immutable
+public final class Point {
+
+    /*
+     * The x-axis point coordinate.
+     */
+    private final float xCoordinate;
+
+    /*
+     * The y-axis point coordinate.
+     */
+    private final float yCoordinate;
+
+    /**
+     * Constructs a Point object.
+     *
+     * @param xCoordinate The x-axis point coordinate.
+     * @param yCoordinate The y-axis point coordinate.
+     */
+    public Point(final float xCoordinate, final float yCoordinate) {
+        this.xCoordinate = xCoordinate;
+        this.yCoordinate = yCoordinate;
+    }
+
+    /**
+     * Gets the x-coordinate value.
+     *
+     * @return The x-axis coordinate value.
+     */
+    public float getX() {
```

Comment:
Wonder if we should use `float` or `double` for these properties. I'd lean towards double as this is meant to be a generic class and double can store float but not vice-versa.